static/app/components/searchQueryBuilder/index.spec.tsx

```diff
@@ -3366,4 +3378,84 @@ describe('SearchQueryBuilder', function () {
       });
     });
   });
+
+  describe('explicitly typed tags', function () {
+    it('renders explicit string tag filter', async function () {
+      render(
+        <SearchQueryBuilder {...defaultProps} initialQuery="tags[foo,string]:foo" />
+      );
+
+      expect(screen.getByLabelText('tags[foo,string]:foo')).toHaveTextContent(
+        /foo.*is.*foo/
```

Comment:
If you want to avoid the regex you could also do:

```
screen.getByRole('button', {name: 'Edit key for filter: tags[foo,string]'}).toHaveTextContent('foo')
```

Reply:
I narrowed the condition and matched on the exact element instead.